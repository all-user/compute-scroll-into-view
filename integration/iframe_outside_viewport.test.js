beforeAll(async () => {
  await page.goto('http://localhost:3000/integration/iframe_outside_viewport')
})

describe('target in iframe is outside viewport', () => {
  test('should scroll top window', async () => {
    expect.assertions(4)
    const actual = await page.evaluate(() => {
      const iframe = document.querySelector('iframe')
      const target = iframe.contentDocument.querySelector('.target')
      return window
        .computeScrollIntoView(target, {
          scrollMode: 'always',
        })
        .map(window.mapActions)
    })
    expect(actual).toHaveLength(2)
    expect(actual[0]).toMatchObject({ el: 'html' })
    expect(actual[1]).toMatchObject({ el: 'html' }, { el: 'html' })
    expect(actual).toMatchSnapshot()
  })
})
