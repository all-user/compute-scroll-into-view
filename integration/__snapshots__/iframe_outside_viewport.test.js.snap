// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`target in iframe is outside viewport should scroll top window 1`] = `
Array [
  Object {
    "el": "html",
    "left": 0,
    "top": 300,
  },
  Object {
    "el": "html",
    "left": 0,
    "top": 124,
  },
]
`;
